wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/nbt/NbtDelegates.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qkl.wrapper.minecraft.nbt
+
+import net.minecraft.nbt.*
+import kotlin.properties.PropertyDelegateProvider
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KCallable
+import kotlin.reflect.KProperty
+
+/**
+ * A property delegate for NBT tags.
+ */
+public typealias NbtProperty<T> = ReadWriteProperty<Any?, T>
+
+/**
+ * A provider for NBT property delegates.
+ */
+public typealias NbtPropertyProvider<T> = PropertyDelegateProvider<Any?, NbtProperty<T>>
+
+/**
+ * A reference to a [Compound][NbtCompound]. To get one, create a variable and reference it with `::variableName`.
+ */
+public typealias CompoundProperty = KCallable<NbtCompound>
+
+internal class Delegate<T>(
+    val compound: CompoundProperty,
```

Comment:
Why use callables and the reference syntax instead of passing the compound directly?

Reply:
Passing references allows for the pointed variable to be updated to provide a new compound, or the return type of a function to be updated each time an operation is done. This essentially makes the delegated references "follow" the compound rather than forever modifying a compound whose contents might not be looked at any longer.